metrics.go

```diff
@@ -45,3 +45,22 @@ func WritePrometheus(w io.Writer, exposeProcessMetrics bool) {
 		writeProcessMetrics(w)
 	}
 }
+
+// WritePrometheusMetricSet will write all metrics registered in the provided set in Prometheus format to w.
+//
+// If exposeProcessMetrics is true, then various `go_*` and `process_*` metrics
+// are exposed for the current process.
+//
+// The WritePrometheusMetricSet func is usually called inside "/metrics" handler:
+//
+//     http.HandleFunc("/metrics", func(w http.ResponseWriter, req *http.Request) {
+//         metrics.WritePrometheusSet(set, w, true)
+//     })
+//
+func WritePrometheusMetricSet(set *Set, w io.Writer, exposeProcessMetrics bool) {
```

Comment:
What is the use case for this method? I think it would be better adding `WriteProcessMetrics(w io.Writer)` function, which writes process metrics to `w`.

Reply:
We are working with our own set of metrics (manually initialized) to be able to use the `Set` operations (like listing and unregistering). This is required because the "defaultSet" is not public and hence I had no access to the `Set` of metrics being used internally.

Adding a function allow writing metrics like `func (s *Set) WriteProcessMetrics(w io.Writer,  exposeProcessMetrics bool)` would work too, I was unsure about that.

Alternatively the private defaultSet could be made public available with a `metrics.GetMetricSet()` method, which would allow to use the existing WriteProcessMetrics function than

Please share your thoughts on this.

Reply:
I don't like the idea with `exposeProcessMetrics` arg, since it hurts composability. This arg has been initially added to [WritePrometheus](https://godoc.org/github.com/VictoriaMetrics/metrics#WritePrometheus) function and later it couldn't be removed because of backwards compatibility.

I think it is better adding the following public function:

```go
func WriteProcessMetrics(w io.Writer) {
  writeGoMetrics(w)
  writeProcessMetrics(w)
}
```

Then this function could be called independently when user needs exporting process metrics. For instance, the `WritePrometheusMetricSet` function could be substituted by the following user code:

```go
s.WritePrometheus(w)
metrics.WriteProcessMetrics(w)
```

Reply:
Added a `WriteProcessMetrics(w io.Writer)` function to the metrics package as suggested.